src/darwin/Framework/CHIP/EndpointDescription/MTRAccessGrant.mm

```diff
@@ -0,0 +1,131 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTRError_Internal.h"
+#import "MTRLogging_Internal.h"
+#import <Matter/MTRAccessGrant.h>
+
+#include <lib/core/CASEAuthTag.h>
+#include <lib/core/GroupId.h>
+#include <lib/core/NodeId.h>
+#include <lib/support/SafeInt.h>
+
+using namespace chip;
+
+@implementation MTRAccessGrant
+
+- (nullable instancetype)initWithNodeID:(NSNumber *)nodeID privilege:(MTRAccessControlEntryPrivilege)privilege error:(NSError * __autoreleasing *)error
+{
+    NodeId id = nodeID.unsignedLongLongValue;
+    if (!IsOperationalNodeId(id)) {
+        MTR_LOG_ERROR("MTRAccessGrant provided non-operational node ID: 0x%llx", id);
+        if (error) {
```

Comment:
I'm a little concerned that we're perpetuating the use of NSError for programmer errors further and further.

Reply:
I am totally open to other options here....  I could just have the initializers return nil without an error outparam; would you prefer that?

Reply:
Switching to not using NSError outparams except in cases where the caller might be able to runtime-recover.